EapWebInterface/src/main/java/de/hpi/unicorn/application/pages/input/generator/EventGenerator.java

```diff
@@ -121,278 +111,53 @@ public void generateEvents(int eventCount, int scaleFactor, EapEventType eventTy
      * If the given attribute can be found as a base attribute in a dependency, this function will set dependency-compliant values for dependent
      * attributes.
      *
-     * @param baseAttribute the attribute to be handled as a base attribute
+     * @param baseAttributeInput contains the concerned event type attribute and its user input
      * @param eventValues a map containing already set values, will set the new values in here too. (Value for base attribute has to be set already!)
      */
-    private void tryToFillDependentAttributes(TypeTreeNode baseAttribute, Map<String, Serializable> eventValues) {
+    private void tryToFillDependentAttributes(AttributeInput baseAttributeInput, Map<String, Serializable> eventValues) {
+        TypeTreeNode baseAttribute = baseAttributeInput.getAttribute();
         if (!attributeDependencyManager.isBaseAttributeInAnyDependency(baseAttribute)) {
             return;
         }
 
-        String baseAttributeInput;
-        if (baseAttribute.getType() == AttributeTypeEnum.DATE) {
-            baseAttributeInput = dateFormatter.format(eventValues.get(baseAttribute.getName()));
-        }
-        else {
-            baseAttributeInput = String.valueOf(eventValues.get(baseAttribute.getName()));
-        }
-
+        // iterate over each dependency defined for the base-attribute
         for (AttributeDependency attributeDependency : attributeDependencyManager.getAttributeDependencies(baseAttribute)) {
             TypeTreeNode dependentAttribute = attributeDependency.getDependentAttribute();
-            List<String> possibleDependentValues = new ArrayList<>();
-            for (AttributeValueDependency attributeValueDependency
-                    : attributeDependencyManager.getAttributeValueDependencies(attributeDependency)) {
-                if (isInRange(baseAttributeInput, attributeValueDependency.getBaseAttributeValue(), baseAttribute.getType())) {
-                    possibleDependentValues.add(attributeValueDependency.getDependentAttributeValues());
+            List<AttributeInput> possibleDependentAttributeInputs = new ArrayList<>();
+            // iterate over each value-dependency defined on this dependency
+            for (AttributeValueDependency attributeValueDependency : attributeDependencyManager.getAttributeValueDependencies
+                    (attributeDependency)) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') '(' should be on the previous line. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://bpt-lab.org/sonarqube/coding_rules#rule_key=checkstyle%3AMethod_Param_Pad_BP2016)